iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ReactiveCapabilityCurveImpl.java

```diff
@@ -54,10 +56,46 @@ public double getMaxQ() {
 
     private final TreeMap<Double, Point> points;
 
-    ReactiveCapabilityCurveImpl(TreeMap<Double, Point> points) {
+    private static void checkPointsSize(TreeMap<Double, Point> points) {
         if (points.size() < 2) {
-            throw new IllegalStateException("Points size must be >= 2");
+            throw new IllegalStateException("points size should be >= 2");
+        }
+    }
+
+    private Point extrapolateReactiveLimitsSlope(double p) {
+        Map.Entry<Double, Point> e1 = points.floorEntry(p);
+        Map.Entry<Double, Point> e2 = points.ceilingEntry(p);
+        double minQ;
+        double maxQ;
+        if (e1 == null && e2 != null) {
+            // Extrapolate reactive limits slope below min active power limit p2
+            Point p2 = e2.getValue();
+            Point p2bis = points.higherEntry(e2.getKey()).getValue(); // p < p2 < p2bis
+            double slopeMinQ = (p2bis.getMinQ() - p2.getMinQ()) / (p2bis.getP() - p2.getP());
+            double slopeMaxQ = (p2bis.getMaxQ() - p2.getMaxQ()) / (p2bis.getP() - p2.getP());
+            minQ = p2.getMinQ() + slopeMinQ * (p - p2.getP());
+            maxQ = p2.getMaxQ() + slopeMaxQ * (p - p2.getP());
+        } else if (e1 != null && e2 == null) {
+            // Extrapolate reactive limits slope above max active power limit p1
+            Point p1 = e1.getValue();
+            Point p1bis = points.lowerEntry(e1.getKey()).getValue(); // p1bis < p1 < p
+            double slopeMinQ = (p1.getMinQ() - p1bis.getMinQ()) / (p1.getP() - p1bis.getP());
+            double slopeMaxQ = (p1.getMaxQ() - p1bis.getMaxQ()) / (p1.getP() - p1bis.getP());
+            minQ = p1.getMinQ() + slopeMinQ * (p - p1.getP());
+            maxQ = p1.getMaxQ() + slopeMaxQ * (p - p1.getP());
+        } else {
+            throw new IllegalStateException();
         }
+        if (minQ <= maxQ) {
+            return new PointImpl(p, minQ, maxQ);
+        } else { // Corner case of intersecting reactive limits when extrapolated
+            double limitQ = (minQ + maxQ) / 2;
+            return new PointImpl(p, limitQ, limitQ); // Returning the intersection as limits
```

Comment:
Maybe add a warning log here ? Idealy with enough detail to troubleshoot and find the limits. And also ideally not to frequently (because OLF will call that many times in a run) . For instance first time it happens for an instance and then every 100 times...